monitor-data-status/update_data_status.py

```diff
@@ -0,0 +1,52 @@
+import synapseclient
+import pandas as pd
+import os
+import json
+import argparse
+
+# Constants for the views
+FILE_VIEW_ID = "syn16858331"
+PROJECT_VIEW_ID = "syn52677631"
+
+def main(dry_run, update_df):
+    syn = synapseclient.Synapse()
+    secrets = json.loads(os.getenv("SCHEDULED_JOB_SECRETS"))
+    auth_token = secrets["SYNAPSE_AUTH_TOKEN"]
+    syn.login(authToken = auth_token)
+
+    if update_df:
+        print(f"Using manually specified data csv...")
+        fileview_df = pd.read_csv(update_df)
+    else:  
+        # Fetch the project view data
+        pending_projects_df = syn.tableQuery(f"SELECT id FROM {PROJECT_VIEW_ID} WHERE dataStatus='Data Pending'").asDataFrame()
+        ids = tuple(pending_projects_df['id'])
+        QUERY_IDS = f"({', '.join(repr(item) for item in ids)})"
+
+        # Fetch the file view data
+        # We'll need to filter out files created by nf-osi service or staff 
+        # who tend to upload data sharing plans & other administrative files
+        print(f"Checking production table for {len(ids)} projects with status 'Data Pending'...")
+        fileview_df = syn.tableQuery(f"SELECT projectId,count(*) as N FROM {FILE_VIEW_ID} WHERE type='file' and createdBy not in ('3421893', '3459953', '3434950', '3342573') and projectId in {QUERY_IDS} group by projectId").asDataFrame()
```

Comment:
```suggestion
        fileview_df = syn.tableQuery(f"SELECT projectId,count(*) as N FROM {FILE_VIEW_ID} WHERE type='file' and createdBy not in ('3421893', '3459953', '3434950', '3342573') and projectId in {QUERY_IDS} group by projectId").asDataFrame()
```
Should we maintain a table on the portal backend project to keep the createdBy principal ids in? Then we could also use that table for reporting (instead of the one listed in the readme here https://github.com/nf-osi/internal-reports)

Reply:
Yeah, we probably need to start a table as a source of truth for who should be excluded for these sorts of things. I think we can do that first and then go back to revise these two + others.